source/extensions/filters/http/local_ratelimit/local_ratelimit.cc

```diff
@@ -0,0 +1,115 @@
+#include "extensions/filters/http/local_ratelimit/local_ratelimit.h"
+
+#include <string>
+#include <vector>
+
+#include "envoy/http/codes.h"
+
+#include "common/http/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace LocalRateLimitFilter {
+
+FilterConfig::FilterConfig(
+    const envoy::extensions::filters::http::local_ratelimit::v3::LocalRateLimit& config,
+    Event::Dispatcher& dispatcher, Stats::Scope& scope, Runtime::Loader& runtime,
+    const bool per_route)
+    : status_(toErrorCode(config.status().code())),
+      stats_(generateStats(config.stat_prefix(), scope)),
+      rate_limiter_(Filters::Common::LocalRateLimit::LocalRateLimiterImpl(
+          std::chrono::milliseconds(
+              PROTOBUF_GET_MS_OR_DEFAULT(config.token_bucket(), fill_interval, 0)),
+          config.token_bucket().max_tokens(),
+          PROTOBUF_GET_WRAPPED_OR_DEFAULT(config.token_bucket(), tokens_per_fill, 1), dispatcher)),
+      runtime_(runtime),
+      filter_enabled_(
+          config.has_filter_enabled()
+              ? absl::optional<Envoy::Runtime::FractionalPercent>(
+                    Envoy::Runtime::FractionalPercent(config.filter_enabled(), runtime_))
+              : absl::nullopt),
+      filter_enforced_(
+          config.has_filter_enabled()
+              ? absl::optional<Envoy::Runtime::FractionalPercent>(
+                    Envoy::Runtime::FractionalPercent(config.filter_enforced(), runtime_))
+              : absl::nullopt),
+      response_headers_parser_(
+          Envoy::Router::HeaderParser::configure(config.response_headers_to_add())) {
+  // Note: no token bucket is fine for the global config, which would be the case for enabling
+  //       the filter globally but disabled and then applying limits at the virtual host or
+  //       route level. At the virtual or route level, it makes no sense to have an no token
+  //       bucket so we throw an error.
```

Comment:
What is the behavior if there is no token bucket configured and the only config is the global one? Can you clarify that in the docs and comment here in the impl?